src/ble/BleLayerObject.h

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2014-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the interface that will be used by objects in BLE Layer.
+ */
+
+#pragma once
+
+namespace chip {
+namespace Ble {
+
+class BleLayer;
+class BleLayerDelegate;
+
+class BleLayerObject
+{
+    friend class BleLayer;
```

Comment:
Why a friend class here?
Why the forward declarations above?

Could we remove the friend declarations to reduce coupling? This enforces that addreff/release is only callable by subclasses or BleLayer. Why this restriction?

Reply:
I found BleLayerObject is only used by BleEndpoint, so I removed it.

Infact, the BleLayerObject is not a new code, I think this can be addressed in future BLE refactor work.